newswires/client/src/SideNav/TopLevelListPreset.tsx

```diff
@@ -22,22 +24,33 @@ export const TopLevelListPresetPanel = ({
 					key={item.id}
 					isActive={
 						activePreset === item.id ||
-						(item.id === 'all-presets' && !activePreset)
+						(item.id === topLevelPresetId && !activePreset)
 					}
 					isTopLevel={true}
 					handleButtonClick={() => {
+						togglePreset(item.id);
 						if (item.child) {
-							swapActivePanel(item.child, 'forward');
-						} else {
-							togglePreset(item.id);
+							if (getNextActivePreset(activePreset, item.id)) {
```

Comment:
there's something a bit hard to parse about this `if/else` block as a reader, although I can't think of a good alternative at the moment so definitely not blocking!